### PR TITLE
refactor clippy-consts to use ConstInt

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: src/consts.rs [hunk 1]
```diff
@@ -92,10 +84,8 @@ impl PartialEq for Constant {
         match (self, other) {
             (&Constant::Str(ref ls, ref lsty), &Constant::Str(ref rs, ref rsty)) => ls == rs && lsty == rsty,
             (&Constant::Binary(ref l), &Constant::Binary(ref r)) => l == r,
-            (&Constant::Byte(l), &Constant::Byte(r)) => l == r,
```
Comment (llogiq) [inline src/consts.rs]: Can Bytes so longer be equal?

**Comment by llogiq** [comment]
Why did you remove Byte?

Otherwise LGTM

File: src/consts.rs (hunk 1 above)
Comment (oli-obk) [inline src/consts.rs]: Bytes have been removed. The are changed to `Constant::Int(ConstInt::U8(...))`. I'm assuming there's no reason to keep bytes around.

File: src/consts.rs [hunk 2]
```diff
@@ -92,10 +84,8 @@ impl PartialEq for Constant {
...
             (&Constant::Char(l), &Constant::Char(r)) => l == r,
-            (&Constant::Int(0, _, _), &Constant::Int(0, _, _)) => true,
-            (&Constant::Int(lv, _, lneg), &Constant::Int(rv, _, rneg)) => lv == rv && lneg == rneg,
+            (&Constant::Int(l), &Constant::Int(r)) => l == r,
```
Comment (mcarton) [inline src/consts.rs]: We used to ignore the type when comparing, not anymore. Couldn’t that lead to false negatives?

**Comment by oli-obk** [comment]
> Why did you remove Byte?

`b'a'` is 100% equal to `97u8`, I don't see any reason to distinguish it.

File: src/consts.rs (hunk 2 above)
Comment (oli-obk) [inline src/consts.rs]: I don't see how... We probably assumed rust caught the type mismatch before we got here.

Comment (mcarton) [inline src/consts.rs]:
```rust
Constant::Int(42, LitIntType::Signed(UintTy::U8), Sign::Plus) ==
Constant::Int(42, LitIntType::Unsuffixed, Sign::Plus);
```
used to be true.
```rust
Constant::Int(ConstInt::Infer(42)) == Constant::Int(ConstInt::U8(42));
```
is not.

Comment (mcarton) [inline src/consts.rs]: Yes, *rustc* will catch `u8 == i64`, but with the `Infer`/`Unsuffixed` thing, what’s happening?

**Comment by mcarton** [comment]
` +66 −233 ` nice cleanup :smile:

File: src/consts.rs (hunk 2 above)
Comment (oli-obk) [inline src/consts.rs]: well... I can change that if we pass a `tcx`. But `Infer` infers to the other type before running any operation

**Comment by llogiq** [comment]
r+ if we can get the unsuffixed/inferred types to work with all other types without tcx (because that's what we'll have in the majority of uses).

File: src/consts.rs (hunk 2 above)
Comment (mcarton) [inline src/consts.rs]: Can’t you just use `erase_type` before you compare/ord/hash?

Comment (oli-obk) [inline src/consts.rs]: oh... I totally forgot about `Hash`... `Infer(5) == U8(5)`, but the hash differs. Comparison will never fail due to type mismatch if either argument is `Infer`, it'll simply convert it. I should probably fix this in rustc.

Comment (oli-obk) [inline src/consts.rs]: Ah no. I'm being stupid... there's no inference for the `PartialEq` *phew*.

**Comment by oli-obk** [comment]
fixed

**Comment by oli-obk** [comment]
travis bumped -> succeeded

**Pull request merged**